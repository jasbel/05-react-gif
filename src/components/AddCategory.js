import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState(" ");

    const handleAdd = (e) => {
        setInputValue(e.target.value);

        console.log("orkinf");
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        //evitar los vacios
        if (inputValue.trim().length > 2) {
            setCategories((cat) => [inputValue, ...cat]);

            //vaciar input
            setInputValue('');
        }
    };

    return (
        <div>
            <h2> {inputValue} </h2>
            <form onSubmit={handleSubmit}>
                <input type="text" value={inputValue} onChange={handleAdd} />
            </form>
            <button onClick={handleSubmit}> Agregar </button>
        </div>
    );
};

// obligar propiedades requeridas
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
