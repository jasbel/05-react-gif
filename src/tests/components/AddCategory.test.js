import React from "react";
import '@testing-library/jest-dom'
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("pruebasa en >AddCategory", () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
      jest.clearAllMocks();
      wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test("debe de mostrarse correctamente", () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('should change boxtext', () => {
      const input = wrapper.find('input');
      const value= 'HOla mundo'

      input.simulate('change', {target: {value}})

      expect(wrapper.find('h2').text().trim()).toBe(value)
    })

    test('should not postear la funcion de submit', () => {

      wrapper.find('form').simulate('submit', {preventDefault(){}})

      expect(setCategories).not.toHaveBeenCalled()
    })

    test('should call setCategories y limpoiaar la caja de texto', () => {

      const value= 'HOla mundo'

      wrapper.find('input').simulate('change', {target: {value}})

      wrapper.find('form').simulate('submit', {preventDefault(){}})

      // console.log(input.prop('value'));

      expect(setCategories).toHaveBeenCalled()
      expect(wrapper.find('input').prop('value').trim()).toBe('')
      
    })

});
