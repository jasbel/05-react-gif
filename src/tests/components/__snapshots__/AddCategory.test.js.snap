// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebasa en >AddCategory debe de mostrarse correctamente 1`] = `
<div>
  <h2>
     
     
     
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=" "
    />
  </form>
  <button
    onClick={[Function]}
  >
     Agregar 
  </button>
</div>
`;
